controls/views.py

```diff
@@ -803,6 +803,46 @@ def import_component(request):
     result = ComponentImporter().import_components_as_json(import_name, oscal_component_json, request)
     return component_library(request)
 
+def statement_history(request, smt_id=None):
+    """Returns the history for the given statement"""
+    from controls.models import Statement
+
+    smt_smt = Statement.objects.get(id=smt_id)
+    full_smt_history = smt_smt.history.all()
+    context = {"statement": full_smt_history}
+
+    return render(request, "controls/statement_history.html", context)
+
+def restore_to_history(request, smt_id, history_id):
+    """
+    Restore the current model instance to a previous version
+    """
+
+    smt = Statement.objects.get(id=smt_id)
+    recent_smt = smt.history.first()
+    historical_smt = smt.history.get(history_id=history_id)
+
+    # saving historical statement as a new instance
+    historical_smt.instance.save()
+    logger.info(f"Restoring the current statement with an id of {smt_id} to version with a history id of {history_id}")
+    messages.add_message(request, messages.INFO, f'Successfully restored the statement to version history {history_id}')
+
+    # Diff between most recent and the historical record
+    all_hist = smt.history.all()
```

Comment:
Instead of calling `smt.history.all()` twice (about 10 lines later, it is called again) and calling `smt.history.first()` 10 lines prior, a single call to `smt.history.all()` should be made at the start and stored. Then you can just call `.first()` on that variable. So like:
`statement_history = smt.history.all()`
`first_statement_history = statement_history.first()`

Then you can use those variables throughout where needed without re-calling those functions. (I'm not sure how much processing it takes to look up a statement history, but it might end up being a lot down the road as statement histories change a lot. So, removing at least one or two calls to those functions might help!